### PR TITLE
Shorten error prefix of custom errors

### LINKED ISSUE #4417: Change error message of custom errors
## Summary

Change: "A custom type error occurred while solving type class constraints" to "Custom error:"

## Motivation

Check [here](https://discourse.purescript.org/t/get-rid-of-a-custom-type-error-occurred-while-solving-type-class-constraints/3215)

### PR DESCRIPTION
**Pull request by i-am-the-slime**
**Description of the change**

Shorten error prefix of custom errors.
Don't think this qualifies as breaking or even for a changelog entry.

Closes #4417 
---

**Checklist:**

- [x] Added a file to CHANGELOG.d for this PR (see CHANGELOG.d/README.md)
- [x] Added myself to CONTRIBUTORS.md (if this is my first contribution)
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

File: tests/purs/failing/2567.out [hunk 1]
```diff
@@ -15,4 +15,3 @@ in value declaration [33mfoo[0m
 
 See https://github.com/purescript/documentation/blob/master/errors/NoInstanceFound.md for more information,
 or to contribute content related to this error.
```
Comment (i-am-the-slime) [inline tests/purs/failing/2567.out:17]:
```suggestion
or to contribute content related to this error.

```

File: tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out [hunk 2]
```diff
@@ -15,4 +15,3 @@ in value declaration [33mmain[0m
 
 See https://github.com/purescript/documentation/blob/master/errors/NoInstanceFound.md for more information,
 or to contribute content related to this error.
```
Comment (i-am-the-slime) [inline tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out:17]:
```suggestion
or to contribute content related to this error.

```

File: tests/purs/failing/ProgrammableTypeErrors.out [hunk 3]
```diff
@@ -25,4 +25,3 @@ where [33mt0[0m is an unknown type
 
 See https://github.com/purescript/documentation/blob/master/errors/NoInstanceFound.md for more information,
 or to contribute content related to this error.
```
Comment (i-am-the-slime) [inline tests/purs/failing/ProgrammableTypeErrors.out:27]:
```suggestion
or to contribute content related to this error.

```

File: tests/purs/failing/ProgrammableTypeErrorsTypeString.out [hunk 4]
```diff
@@ -22,4 +22,3 @@ where [33mt0[0m is an unknown type
 
 See https://github.com/purescript/documentation/blob/master/errors/NoInstanceFound.md for more information,
 or to contribute content related to this error.
```
Comment (i-am-the-slime) [inline tests/purs/failing/ProgrammableTypeErrorsTypeString.out:24]:
```suggestion
or to contribute content related to this error.

```

**Comment by rhendric** [comment]
It's a user-visible change, so it should go in the changelog. It's a `feature`, albeit a very small one.

File: CHANGELOG.d/feature_shorten-error-message.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+* `feature`: shorten the prefix for custom user defined error
+  messages to improve clarity and get to the relevant information
+  more quickly
```
Comment (purefunctor) [inline CHANGELOG.d/feature_shorten-error-message.md:3]: Could you reword this to be:
```
* Shorten the prefix [...]
```

**Review by purefunctor — APPROVED** [review]
LGTM aside from a short nit about the changelog entry. Thanks!

File: CHANGELOG.d/feature_shorten-error-message.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+* `feature`: shorten the prefix for custom user defined error
```
Comment (i-am-the-slime) [inline CHANGELOG.d/feature_shorten-error-message.md:1]:
```suggestion
* `feature`: Shorten the prefix for custom user defined error
```

**Review by JordanMartinez — APPROVED** [review]

File: CHANGELOG.d/feature_shorten-error-message.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+* `feature`: Shorten the prefix for custom user defined error
```
Comment (rhendric) [inline CHANGELOG.d/feature_shorten-error-message.md:1]:
```suggestion
* Shorten the prefix for custom user defined error
```
